src/tests/ftest/util/write_host_file.py

```diff
@@ -43,6 +44,29 @@ def write_host_file(hostlist, path='/tmp', slots=1):
         str: the full path of the written hostfile
 
     """
+
+    # Adjust slot parameter if we get duplicate entries of a hostname, which
+    # will prevent the generation of hostfiles that would throw this orterun
+    # error:
+    #   "A hostfile was provided that contains multiple definitions of the slot
+    #   count for at least one node."
+    #
+    # In other words, this is a valid hostfile:
+    # boro-A
+    # boro-A
+    #
+    # This is a valid hostfile:
+    # boro-A slots=2
+    #
+    # But this is *not* a valid hostfile:
+    # boro-A slots=1
+    # boro-A slots=1
+
+    h = hostlist
+    hosts_with_multiple_entries = [k for k,v in Counter(h).items() if v > 1]
+    if len(hosts_with_multiple_entries):
+        slots = None
+
```

Comment:
The caller of this method shouldn't be responsible for passing in the correct argument.  I have no problem with this method checking the argument and raising an error, e.g.
```
        if hostlist != list(set(hostlist)):
            raise AttributeError(
                "Cannot create a hostfile with duplicate hosts: {}".format(hostlist))
```

If you then want to avoid the error when calling the method, just use `set` to remove any duplicates.

Reply:
Caller of `write_host_file` is now responsible for passing in the correct argument: 6267ef102ca41184c0c4c801dd713ee19fe4cf2d.